File: Template/Naked Objects Server/Template.Model/TeacherRepository.cs
using NakedObjects;
using System.Linq;

namespace Template.Model
{
    public class TeacherRepository
    {
        #region Injected Services
        //An implementation of this interface is injected automatically by the framework
        public IDomainObjectContainer Container { set; protected get; }
        #endregion

        public Teacher CreateNewTeacher() => Container.NewTransientInstance<Teacher>();

        public IQueryable<Teacher> AllTeachers() =>  Container.Instances<Teacher>();

        public IQueryable<Teacher> FindTeacherByName(string name) =>  AllTeachers().Where(c => c.FullName.ToUpper().Contains(name.ToUpper()));
    }

}

File: Template/NOF2 Server/Model.Template/TeacherRepository.vb





Public Class TeacherRepository
        Public Property Container As IContainer

        Public Function CreateNewTeacher() As Teacher
        Return Container.CreateTransientInstance(Of Teacher)()
    End Function

        Public Function AllTeachers() As IQueryable(Of Teacher)
        Return Container.AllInstances(Of Teacher)()
    End Function

        Public Function FindTeacherByName(ByVal name As String) As IQueryable(Of Teacher)
            Return AllTeachers().Where(Function(c) c.FullName.ToUpper().Contains(name.ToUpper()))
        End Function
    End Class

